### PR TITLE
Implements task logging verbosity configuration options.

### PR DESCRIPTION
**Pull request by dangerdespain**
See the config file for the user-facing updates to task logging. The verbose option was left intact for scheduler logging, and is used to override the task verbosity. I did this to keep a single option around for quick debugging but I'm open to tweaking this!

This change is backwards-compatible and transparent to extisting configurations and includes documentation updates (Implements #719)

### CODE REVIEW

File: initializers/exceptions.js [hunk 1]
```diff
@@ -98,7 +98,7 @@ module.exports = {
         simpleName = err.message;
       }
       var name = 'task:' + simpleName;
-      api.exceptionHandlers.report(err, 'task', name, {task: task, queue: queue, workerId: workerId}, 'error');
+      api.exceptionHandlers.report(err, 'task', name, {task: task, queue: queue, workerId: workerId}, api.config.tasks.verbose ? 'info' : api.config.tasks.taskLogging.success);
```
Comment (evantahler) [inline initializers/exceptions.js]: This should be `failure`?

Comment (dangerdespain) [inline initializers/exceptions.js]: Yes, good catch! Copypasta.. Fix inbound.

File: site/source/includes/docs/core/tasks.md [hunk 2]
```diff
@@ -55,6 +55,14 @@ exports.default = {
       scheduler: false,
       // what queues should the TaskProcessors work?
       queues: ['*'],
+      // Verbosity of task scheduler (overrides the below task logging levels when set to true)
+      verbose: true,
+      // Logging levels of tasks
+      taskLogging : {
+        start : 'info',
+        success: 'info',
+        failure: 'error',
```
Comment (evantahler) [inline site/source/includes/docs/core/tasks.md]: I think you need options for **all** possible emitters: `poll`, `reEnqueue`, etc.

File: site/source/includes/docs/core/tasks.md [hunk 3]
```diff
@@ -55,6 +55,14 @@ exports.default = {
       scheduler: false,
       // what queues should the TaskProcessors work?
       queues: ['*'],
+      // Verbosity of task scheduler (overrides the below task logging levels when set to true)
+      verbose: true,
```
Comment (evantahler) [inline site/source/includes/docs/core/tasks.md]: Why does the scheduler have a separate pattern?  It should match the worker and have a hash.

File: site/source/includes/docs/core/tasks.md (hunk 2 above)
Comment (dangerdespain) [inline site/source/includes/docs/core/tasks.md]: I was keeping things simple and leaving the debug-level stuff alone. We can take it to that level of detail though, if you'd like.

Comment (evantahler) [inline site/source/includes/docs/core/tasks.md]: I would! it's more about having different patters for some emitters and not others.

**Comment by dangerdespain** [comment]
So should the verbose option be stripped altogether, then? I like the idea of having a single override for debugging, but perhaps it's just too confusing.

**Comment by evantahler** [comment]
I agree it should be removed

**Comment by dangerdespain** [comment]
Alright.. you may still not like having all of these options under the taskLogging object. If that's the case, I'd like more specific direction on how you'd like the options split between taskLogging and schedulerLogging and would be happy to make those changes.

**Comment by evantahler** [comment]
This is great! 
I think for the scheduler, you just make another hash, schedulerLogging, which has its events as well.  Yes, this makes the config file pretty large, but now we have *all* the options exposed to user-space, which is nice.

File: config/tasks.js [hunk 4]
```diff
@@ -21,7 +32,7 @@ exports.default = {
       // When we kill off a taskProcessor, should we disconnect that local redis connection?
       toDisconnectProcessors: true,
       // What redis server should we connect to for tasks / delayed jobs?
-      redis: api.config.redis
+      redis: api.config.redis,
```
Comment (l0oky) [inline config/tasks.js]: Is that extra comma intentionally left there? `,` :smiley:

File: config/tasks.js [hunk 5]
```diff
@@ -5,8 +5,19 @@ exports.default = {
...
+        start : 'info',
+        end : 'info',
+        success: 'info',
+        failure: 'error',
+        cleaning_worker : 'info',
+        poll : 'debug',
+        job : 'debug',
+        enqueue: 'debug',
+        reEnqueue: 'debug',
+        pause: 'debug',
```
Comment (l0oky) [inline config/tasks.js]: Trailing comma

**Comment by evantahler** [comment]
Wait... why are trailing commas so bad?

**Comment by neilstuartcraig** [comment]
It's invalid JSON, strictly speaking.
That graphic is awesome :-)

**Comment by dangerdespain** [comment]
Haha you guys crack me up.

I'm coming down from a weekend in the mountains, should have this PR wrapped up tomorrow (:

**Comment by dangerdespain** [comment]
Alright, my apologies for the delay. This ought to do the trick.The failing build seems to be unrelated timeouts.

Please be extra verbose if further tweaks are needed ;)

**Comment by evantahler** [comment]
Yeah, sometimes travis times out.  I re-ran them and it's all green!

Thanks!

**Pull request merged**